homeassistant/components/ecoal_boiler.py

```diff
@@ -0,0 +1,46 @@
+"""
+Component to control ecoal/esterownik.pl coal/wood boiler controller.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/ecoal_boiler/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+
+REQUIREMENTS = ['ecoaliface==0.4.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "ecoal_boiler"
+DATA_ECOAL_BOILER = 'data_' + DOMAIN
+
+CONF_USERNAME = "username"
+DEFAULT_USERNAME = "admin"
+DEFAULT_PASSWORD = "admin"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_USERNAME,
+                     default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD,
+                     default=DEFAULT_PASSWORD): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up global ECoalController instance same for sensors and switches."""
+    from ecoaliface.simple import ECoalController
+
+    conf = config.get(DOMAIN)
+    host = conf.get(CONF_HOST)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.

Reply:
Done.